### PR TITLE
Skip computation incase of Null Literal for case/when transformation

### PR DESCRIPTION
**Pull request by abhioncbr**
As per the discussion of the [PR](https://github.com/apache/pinot/pull/11395) 

- Skipping computation of value block in case of Null Literal for case/when transformation.

cc @shenyu0127 @Jackie-Jiang

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java [hunk 1]
```diff
@@ -128,16 +129,22 @@ private void constructStatementListLegacy(List<TransformFunction> arguments) {
...
       _elseStatement = arguments.get(arguments.size() - 1);
     }
   }
 
   private TransformResultMetadata calculateResultMetadata() {
-    TransformResultMetadata elseStatementResultMetadata = _elseStatement.getResultMetadata();
-    DataType dataType = elseStatementResultMetadata.getDataType();
-    Preconditions.checkState(elseStatementResultMetadata.isSingleValue(),
-        "Unsupported multi-value expression in the ELSE clause");
+    DataType dataType;
```
Comment (abhioncbr) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java:138]: This change is required for supporting the NULL `_elseStatement`. Setting it up as `DataType.UNKNOWN`

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java [hunk 2]
```diff
@@ -257,6 +264,14 @@ public TransformResultMetadata getResultMetadata() {
     return _resultMetadata;
   }
 
+  private boolean isNotNullLiteralTransformation(TransformFunction function) {
+    if (Objects.equals(function.getName(), LiteralTransformFunction.FUNCTION_NAME)) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java:268]:
```suggestion
    if (function instanceof LiteralTransformFunction) {
```

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java [hunk 3]
```diff
@@ -257,6 +264,14 @@ public TransformResultMetadata getResultMetadata() {
...
+      LiteralTransformFunction literalFUnction = (LiteralTransformFunction) function;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java:269]: (typo)
```suggestion
      LiteralTransformFunction literalFunction = (LiteralTransformFunction) function;
```

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java:138]: Good catch

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java [hunk 4]
```diff
@@ -128,16 +129,22 @@ private void constructStatementListLegacy(List<TransformFunction> arguments) {
...
   }
 
   private TransformResultMetadata calculateResultMetadata() {
-    TransformResultMetadata elseStatementResultMetadata = _elseStatement.getResultMetadata();
-    DataType dataType = elseStatementResultMetadata.getDataType();
-    Preconditions.checkState(elseStatementResultMetadata.isSingleValue(),
-        "Unsupported multi-value expression in the ELSE clause");
+    DataType dataType;
+    if (_elseStatement == null) {
+      // only possible in case of Null Literal transformation.
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java:140]: This comment is not accurate. I believe the existing code has a bug that can throw `NPE` when there is no else statement. Suggest just removing this comment

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java [hunk 5]
```diff
@@ -269,4 +269,9 @@ public RoaringBitmap getNullBitmap(ValueBlock valueBlock) {
     bitmap.add(0L, length);
     return bitmap;
   }
+
+  // checks whether the literal is null or not.
```
Comment (shenyu0127) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java:273]: nit: The comment is redundant, it simply repeats what the function name says.

https://blog.boot.dev/clean-code/code-comments/
Section "AVOID REDUNDANT COMMENTS. STRIVE FOR A SINGLE SOURCE OF TRUTH"

**Review by shenyu0127 — COMMENTED** [review]
If would be good to add a test to `CaseTransformFunctionTest` which uses a query "SELECT CASE WHEN xx THEN xx ELSE NULL" and assert the query results to exercise the modified code paths.

**Comment by shenyu0127** [comment]
> If would be good to add a test to `CaseTransformFunctionTest` which uses a query "SELECT CASE WHEN xx THEN xx ELSE NULL" and assert the query results to exercise the modified code paths.

Did you add a test? The rest looks good to me.

**Pull request merged**